File: web/src/plugin/mock_subscribe/api/contract.js
import service from '@/utils/request'

export const getContractList = (params) => {
  return service({
    url: '/mockSubscribeContract/getContractList',
    method: 'get',
    params
  })
}

export const findContract = (params) => {
  return service({
    url: '/mockSubscribeContract/findContract',
    method: 'get',
    params
  })
}

export const updateContractStatus = (data) => {
  return service({
    url: '/mockSubscribeContract/updateContractStatus',
    method: 'put',
    data
  })
}

export const getContractRecordList = (params) => {
  return service({
    url: '/mockSubscribeContract/getContractRecordList',
    method: 'get',
    params
  })
}

File: server/plugin/mock_subscribe/router/contract.go
package router

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type contract struct{}

func (r *contract) Init(public *gin.RouterGroup, private *gin.RouterGroup) {
	_ = public
	group := private.Group("mockSubscribeContract").Use(middleware.OperationRecordWithUserID(3))
	group.GET("getContractList", apiInfo.Contract.GetContractList)
	group.GET("findContract", apiInfo.Contract.FindContract)
	group.PUT("updateContractStatus", apiInfo.Contract.UpdateContractStatus)
	group.GET("getContractRecordList", apiInfo.Contract.GetContractRecordList)
}
